src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -812,6 +831,38 @@ private static bool CanReplace(ISymbol symbol)
             return memberAccess.CanReplaceWithReducedName(replacementNode, semanticModel, cancellationToken);
         }
 
+        /// <summary>
+        /// Determines if <paramref name="speculativeSymbol"/> contains a superset of the symbols in
+        /// <paramref name="actualSymbol"/>.
+        /// </summary>
+        private static bool IsReplacementCandidate(SymbolInfo actualSymbol, SymbolInfo speculativeSymbol)
```

Comment:
Overall, shouldn't we only be removing this/base if the name after hte dot binds without errors?  I would think we would only do this if and only if actualSymbl and speculativeSymbol have ".Symbol" set and not if they have Candidates set.